dashboard/lib/tasks/setup_or_migrate.rake

```diff
@@ -9,8 +10,10 @@ namespace :db do
     true
   end
 
-  task :setup_or_migrate => [:environment, :load_config] do
-    if database_exists?
+  task :setup_or_migrate do
+    db_exists = database_exists?
+    Rake::Task["db:load_config"].invoke
```

Comment:
Do I understand correctly that you've moved `db:load_config` out of the task dependencies to preserve task ordering (make sure it runs after `environment`)?

That seems over-cautious here.  Either `load_config` depends on `environment`, in which case `load_config` also belongs inside the `database_exists?` check, or it doesn't, in which case it ought to be okay to run first (and leave it as a task dependency).

Unless you have a different (code style?) reason for doing it this way.